ui/src/AzExtOutputChannel.ts

```diff
@@ -0,0 +1,50 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { OutputChannel, ViewColumn } from "vscode";
+
+export class AzExtOutputChannel implements OutputChannel {
```

Comment:
Rather than `implements OutputChannel` do `implements types.AzExtOutputChannel`

with 
```
import * as types from '../../index';
```

That helps force you to keep the class itself up to date with the typing in index.d.ts